src/widget/wlibrarysidebar.cpp

```diff
@@ -239,12 +239,36 @@ void WLibrarySidebar::keyPressEvent(QKeyEvent* event) {
     // TODO(XXX) Should first keyEvent ensure previous item has focus? I.e. if the selected
     // item is not focused, require second press to perform the desired action.
 
-    // make the selected item the navigation starting point
+    SidebarModel* sidebarModel = qobject_cast<SidebarModel*>(model());
+    QModelIndexList selectedIndices = selectionModel()->selectedRows();
+    if (sidebarModel && !selectedIndices.isEmpty()) {
+        QModelIndex index = selectedIndices.at(0);
+        if (event->matches(QKeySequence::Delete) || event->key() == Qt::Key_Backspace) {
+            sidebarModel->clear(index);
+            return;
+        }
+        if (event->matches(QKeySequence::Cut)) {
+            sidebarModel->cut(index);
+            return;
+        }
+        if (event->matches(QKeySequence::Copy)) {
+            sidebarModel->copy(index);
+            return;
+        }
+        if (event->matches(QKeySequence::Paste)) {
+            sidebarModel->paste(index);
+            return;
+        }
+        if (event->matches(QKeySequence::SelectAll)) {
+            sidebarModel->selectAll(index);
```

Comment:
This does nothing

Reply:
All methods that are empty or only called empty methods have been removed. Please resolve this conversation, @daschuer